tests/post/steps/test_monitoring.py

```diff
@@ -174,6 +182,69 @@ def _node_has_metrics():
     utils.retry(_node_has_metrics, times=60, wait=3)
 
 
+@then(
+    "the deployed Prometheus alerting rules is the same as the default "
+    "alerting rules")
+def monitor_deployed_rules(host):
+    """
+        This test is meant to be a guard to catch Prometheus-operator charts
+        upgrade changes that might introduce new alerting rules which warrant
+        some documentation efforts.
+
+        Only a Prometheus alerting rule change should trigger this to fail.
+        When this happens, Developers need to regenerate the default alerting
+        rules using the rule_extractor python script, compare and document the
+        new changes in the Cluster monitoring guide before moving on.
+    """
+    def _wait_prom_rule_cmp():
+        response = _query_prometheus_api(host, 'rules')
+        assert response.status_code == 200, \
+            "Expect status code <200> but got {} for route rules".format(
+                response.status_code
+            )
+        deployed_rules = response.json()
+
+        # we cannot simply compare the obtained value because it contains
+        # alerts with an "activeAt": "2020-05-15T09:23:15.279979961Z",
+        # so lets filter and use only certain alerting rule fields
+        rule_group = deployed_rules.get('data', {}).get('groups', [])
+        deployed_alerting_rules = []
+        for item in rule_group:
+            for rule in item.get('rules', []):
+                # rule type can be alerting or recording
+                # For now, we only need alerting rules
+                if rule['type'] == "alerting":
+                    message = rule['annotations'].get('message') or \
+                        rule['annotations'].get('summary')
+                    fixup_alerting_rule = {
+                        'name': rule['name'],
+                        'severity': rule['labels']['severity'],
+                        'message': message
+                    }
+                    deployed_alerting_rules.append(fixup_alerting_rule)
+
+        filename = 'alerting_rules.json'
+        filepath = (pathlib.Path(__file__)/'..'/'..'/'..'/'..'
+                    /'tools'/'rule_extractor'/filename).resolve()
+
+        try:
+            with open(filepath) as f:
+                default_alerting_rules = json.load(f)
+        except IOError as exc:
+            pytest.fail("Failed to open file {}: {!s}".format(filename, exc))
+
+        assert json.dumps(default_alerting_rules, sort_keys=True) \
+            == json.dumps(deployed_alerting_rules, sort_keys=True), \
```

Comment:
That works, but I'd suggest you use Python dicts equality, `pytest` automatically displays the diff when the assertion fails.

If you really want to compare a sorted list, you can do:
```python
sorted(
    default_alerting_rules,
    key=lambda rule: hash(json.dumps(rule, sort_keys=True))
)
```

But that's way more complex than required. Instead, just build a hierarchy of group/alerts and compare the final dict:

```python
# this is an example, adapt it to your code
deployed_rules = {
    group['name']: {rule['name']: rule for rule in group.rules}
    for group in prom_data['data']['groups']
}
```